interpreter/cling/lib/Interpreter/NullDerefProtectionTransformer.cpp

```diff
@@ -203,24 +171,31 @@ namespace cling {
           if (ArgIndexs.test(index)) {
             // Get the argument with the nonnull attribute.
             Expr* Arg = CE->getArg(index);
+            NodeContext newArg = SynthesizeCheck(Arg->getLocStart(), Arg);
+            if (newArg.isSingleStmt()) {
+              assert ((clang::Expr*)newArg.getStmt()
+                      && "Visiting Stmt which is NodeContext an Expr!");
             CE->setArg(index,
-                      (clang::Expr*)SynthesizeCheck(Arg->getLocStart(), Arg));
+                      (clang::Expr*)newArg.getStmt());
+            }
           }
         }
       }
-      return result;
+
+      return NodeContext(CE);
     }
 
     NodeContext VisitCXXMemberCallExpr(CXXMemberCallExpr* CME) {
       Expr* Callee = CME->getCallee();
       if (isa<MemberExpr>(Callee)) {
-        NodeContext ME = Visit(Callee);
-        if (!ME.isSingleStmt())
-          CME->setCallee((clang::Expr*)ME.getStmt());
+        NodeContext newME = Visit(Callee);
+        if (newME.isSingleStmt())
+          assert ((clang::Expr*)newME.getStmt()
+                      && "Visiting Stmt which is not an Expr!");
+          CME->setCallee((clang::Expr*)newME.getStmt());
       }
 
-      NodeContext result(CME);
-      return result;
+      return NodeContext(CME);
```

Comment:
Do we still need this function? I don't see it calling `SynthesizeCheck()` - so it should be covered by `VisitStmt()`, right?